src/cryptography/hazmat/asn1/asn1.py

```diff
@@ -109,7 +109,7 @@ def sequence(cls: type[U]) -> type[U]:
         # parameters.
         dataclass_cls = dataclasses.dataclass(
             repr=False,
-            eq=False,
+            eq=True,
```

Comment:
Do we need this? Could we just have the test types implement `__eq__` (perhaps with teh dataclass decorator themselves?)

Reply:
Do you mean something like this?:
```python
        @asn1.sequence
        @dataclass(eq=True)
        class Example:
            foo: int
```

I'm a bit worried that we would be testing classes that are in some way different than what users would use.

We could declare the dataclass here using the same parameters as the one in `sequence` (except for `eq`), so that no new behavior is added compared to the plain `@sequence` dataclass, but that requires the same version-dependent logic we implement [here](https://github.com/pyca/cryptography/blob/31edc05f7cbb77d2d39ca737e51838794bfd2588/src/cryptography/hazmat/asn1/asn1.py#L79-L117).

Is there a strong reason to not have the Sequence dataclasses with equality comparison by default?

Reply:
Something like that, yes.

The reason not to do have all sequences implement eq is simply that we don't know what users would like, and our goal isn't to build dataclasses, it's to build an ASN.1 library :-)

Reply:
fixed, now it looks like:
```python
        @asn1.sequence
        @_comparable_dataclass
        class Example:
            foo: int
```

where `_comparable_dataclass` contains the version-specific logic